test/examples_test.go

```diff
@@ -0,0 +1,236 @@
+// +build e2e,examples
+
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"io/ioutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	knativetest "knative.dev/pkg/test"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+const (
+	DEFAULT_KO_DOCKER_REPO = `gcr.io\/christiewilson-catfactory`
+	DEFAULT_NAMESPACE      = `namespace: default`
+)
+
+// GetCreatedTektonCrd parses output of an external ko invocation provided as
+// input, as is the kind of Tekton CRD to search for (ie. taskrun)
+func GetCreatedTektonCrd(input []byte, kind string) (string, error) {
+	re := regexp.MustCompile(kind + `.tekton.dev\/(.+) created`)
+	submatch := re.FindSubmatch(input)
+	if submatch == nil || len(submatch) < 2 {
+		return "", nil
+	}
+	return string(submatch[1]), nil
+}
+
+func waitValidatePipelineRunDone(t *testing.T, c *clients, pipelineRunName string) {
+	err := WaitForPipelineRunState(c, pipelineRunName, pipelineRunTimeout, Succeed(pipelineRunName), pipelineRunName)
+
+	if err != nil {
+		t.Fatalf("Failed waiting for pipeline run done: %v", err)
+	}
+	return
+}
+
+func waitValidateTaskRunDone(t *testing.T, c *clients, taskRunName string) {
+	// Per test basis
+	err := WaitForTaskRunState(c, taskRunName, Succeed(taskRunName), taskRunName)
+
+	if err != nil {
+		t.Fatalf("Failed waiting for task run done: %v", err)
+	}
+	return
+}
+
+// SubEnv substitutes docker repos and bucket paths from the system environment
+// for input to allow tests on local clusters. It also updates the namespace
+// for ServiceAccounts so that they work under test
+func SubEnv(input []byte, namespace string) ([]byte, error) {
+	val, ok := os.LookupEnv("KO_DOCKER_REPO")
+	var output []byte
+	if ok {
+		re := regexp.MustCompile(DEFAULT_KO_DOCKER_REPO)
+		output = re.ReplaceAll(input, []byte(val))
+	} else {
+		output = input
+	}
+
+	re := regexp.MustCompile(DEFAULT_NAMESPACE)
+	output = re.ReplaceAll(output, []byte(strings.ReplaceAll(DEFAULT_NAMESPACE, "default", namespace)))
+	return output, nil
+}
+
+// KoCreate wraps the ko binary and invokes `ko create` for input within
+// namespace
+func KoCreate(input []byte, namespace string) ([]byte, error) {
+	cmd := exec.Command("ko", "create", "-n", namespace, "-f", "-")
```

Comment:
I am still not sure why we are using `ko` instead of `kubectl` (we can change that in a follow-up)